frontend/src/components/StudentHistory/Timeline.vue

```diff
@@ -0,0 +1,132 @@
+<template>
+    <div class ="Timeline">
+        <div class="TimelineItem" v-for="(item, index) in items" :key="index">
+                <div id="date"> {{item.date}} </div>
+                <!-- Length of vertical line changes based on position in array-->
+                <div class="firstline" v-if="index == 0"></div>
+                <div class="lastline" v-else-if="index == items.length - 1"></div>
+                <div class ="line" v-else></div>
+                <!-- If a student has dropped out, white dot is rendered -->
+                <div class="whitedot" v-if="item.status == 3"></div>
+                <div class ="dot" v-else></div>
+                <div id="description"> {{item.description}} </div>
+        </div>
+    </div>
+
+</template>
+
+<script>
+export default {
+    name: "Timeline",
+    props: {
+        items: {
+            type: Array,
+            default: () => []
+            /*
+            * item: { date: Date, description: String, status: Number }
+            */ 
+        }
+    }
+    
+    
+}
+</script>
+
+<style>
+.TimelineItem {
+/* Blue Rectangle Background */
+    border: 3px solid white;
+    width: 384px;
+    height: 56px;
+    left: 1000px;
+    top: 410px;
+    
+/* Primary Light */
+
+    background: #F0FCFF;
+    border-radius: 4px;
+    
+/*Font styling*/
+    font-size: 14px;
+    line-height: 20px;
+
+/* identical to box height, or 143% */
+    display: flex;
+    align-items: center;
+
+/* Font colour: Primary Dark */
+    color: #2F4858; 
+    }
+
+
+#date {
+    /*border: 1px solid green;*/
+    margin: auto;
+    margin-right: 9px;
+    width: 85px; /* deviated from Figma*/
+    height: 20px;
+    
+}
+#description {
+   margin-top: 8px;
+   margin-bottom: 8px;
+   margin-right: 24px;
+   margin-left: 9px;
+   width: 225px;
+   height: 40px;
```

Comment:
Do we need an explicit height here? The text might break into 2 lines (e.g. if a name is long).

Reply:
there's an explicit height as the vertical line running through the dots depends on the height of the elements being the same — as opposed to one big line running through all entries, each timeline item has a small line running through the dot, which joins together with other lines for other timeline items to form one long line. 
Guess we wouldn't need explicit height if the lines were dynamic, but I haven't figured out how.